cmake/stm32/utilities.cmake

```
@@ -90,66 +102,59 @@ function(stm32_fetch_cube)
-            message(INFO "STM32_CUBE_${FAMILY}_PATH specified, skipping fetch for ${CUBE_NAME}")
+            message(VERBOSE "STM32_CUBE_${FAMILY}_PATH specified, skipping fetch for ${CUBE_NAME}")
             continue()
         endif()
         
-		FetchContent_GetProperties(${CUBE_NAME} POPULATED CUBE_POPULATED)
-        if(NOT CUBE_POPULATED)
-            set(FETCHCONTENT_QUIET FALSE) # To see progress
-            FetchContent_Populate(${CUBE_NAME})
-        endif()
-        
+		FetchContent_MakeAvailable(${CUBE_NAME})
         set(STM32_CUBE_${FAMILY}_PATH ${${CUBE_NAME_L}_SOURCE_DIR} PARENT_SCOPE)
     endforeach()
 endfunction()
 
 function(stm32_fetch_cmsis)
 	if(NOT STM32_CMSIS_PATH)
-        if(NOT STM32-CMSIS_POPULATED)
-            set(FETCHCONTENT_QUIET FALSE) # To see progress
-            FetchContent_Populate(STM32-CMSIS)
-        endif()
-        
+        FetchContent_MakeAvailable(STM32-CMSIS)
         set(STM32_CMSIS_PATH ${stm32-cmsis_SOURCE_DIR} PARENT_SCOPE)
 	else()
 		message(INFO "STM32_CMSIS_PATH specified, skipping fetch for STM32-CMSIS")
 	endif()
+
     foreach(FAMILY ${ARGV})
-        set(CMSIS_NAME STM32-CMSIS-${FAMILY})
-        string(TOLOWER ${CMSIS_NAME} CMSIS_NAME_L)
-        
-        if(STM32_CMSIS_${FAMILY}_PATH)
-            message(INFO "STM32_CMSIS_${FAMILY}_PATH specified, skipping fetch for ${CMSIS_NAME}")
-            continue()
-        endif()
-        
-		FetchContent_GetProperties(${CMSIS_NAME_L} POPULATED CMSIS_POPULATED)
-        if(NOT CMSIS_POPULATED)
-            set(FETCHCONTENT_QUIET FALSE) # To see progress
-            FetchContent_Populate(${CMSIS_NAME})
+        if(STM32_USE_CMSIS_FROM_CUBE_${FAMILY})
+            stm32_fetch_cube(${FAMILY})
+            set(STM32_CMSIS_${FAMILY}_PATH ${STM32_CUBE_${FAMILY}_PATH}/Drivers/CMSIS/Device/ST/STM32${FAMILY}xx PARENT_SCOPE)
+        else()
+            set(CMSIS_NAME STM32-CMSIS-${FAMILY})
+            string(TOLOWER ${CMSIS_NAME} CMSIS_NAME_L)
+            
+            if(STM32_CMSIS_${FAMILY}_PATH)
+                message(INFO "STM32_CMSIS_${FAMILY}_PATH specified, skipping fetch for ${CMSIS_NAME}")
+                continue()
+            endif()
+            
+            FetchContent_MakeAvailable(${CMSIS_NAME})            
+            set(STM32_CMSIS_${FAMILY}_PATH ${${CMSIS_NAME_L}_SOURCE_DIR} PARENT_SCOPE)
         endif()
-        
-        set(STM32_CMSIS_${FAMILY}_PATH ${${CMSIS_NAME_L}_SOURCE_DIR} PARENT_SCOPE)
     endforeach()
 endfunction()
 
 function(stm32_fetch_hal)
     foreach(FAMILY ${ARGV})
-        set(HAL_NAME STM32-HAL-${FAMILY})
-        string(TOLOWER ${HAL_NAME} HAL_NAME_L)
-        
-        if(STM32_HAL_${FAMILY}_PATH)
-            message(INFO "STM32_HAL_${FAMILY}_PATH specified, skipping fetch for ${HAL_NAME}")
-            continue()
+        if(STM32_USE_HAL_FROM_CUBE_${FAMILY})
+            stm32_fetch_cube(${FAMILY})
+            message("Cube fetched for ${FAMILY} at ${STM32_CUBE_${FAMILY}_PATH}")

```

Please also add the message for CMSIS to stay coherent
```suggestion
            message(STATUS "Cube fetched for ${FAMILY} at ${STM32_CUBE_${FAMILY}_PATH}")
```

---

Ok